common/stream-ratelimiter/src/lib.rs

```diff
@@ -62,7 +62,7 @@ impl<T: Stream> RateLimiter<T> {
             inner: stream,
             interval: interval(refill_interval).fuse(),
             // Start with 0 permits because tokio::interval ticks immediately on setup.
-            permits: Semaphore::new(0),
+            permits: Arc::new(Semaphore::new(0)),
```

Comment:
Does the first fuzzing change depend on this commit to succeed?  If so, let's put it as the first commit to fix the bug :)

Reply:
it doesn't fail, just leaks some bytes : p